scripts/xtensa-build-zephyr.py

```diff
@@ -302,22 +283,60 @@ def check_west_installation():
 
-	:return: [description] Returns path when west workspace had been found, otherwise None
-	:rtype: [type] bool, pathlib.Path
+	:param west_root_dir: directory where is initialized.
+	:type west_root_dir: pathlib.Path
+	:param west_manifest_path: manifest file to which west is initialized.
+	:type west_manifest_path: pathlib.Path
+	:raises RuntimeError: Raised when west is initialized to wrong manifest file
+	(not SOFs manifest) and script is running in non-interactive mode.
+	"""
+	global west_top
+	message = "West is initialized to manifest other than SOFs!\n"
+	message +=  f"Initialized to manifest: {west_manifest_path}." + "\n"
+	dot_west_directory  = pathlib.Path(west_root_dir.resolve(), ".west")
+	if args.no_interactive:
+		message += f"Try deleting {dot_west_directory } directory and rerun this script."
+		raise RuntimeError(message)
+	question = message + "Reinitialize west to SOF manifest? [Y/n] "
+	print(f"{question}")
+	while True:
+		reinitialize_answer = input().lower()
+		if reinitialize_answer == "y" or reinitialize_answer == "n":
+			break
+		sys.stdout.write('Please respond with \'Y\' or \'n\'.\n')
+
+	if not reinitialize_answer:
+		sys.exit("Can not proceed. Reinitialize your west manifest to SOF and rerun this script.")
+	shutil.rmtree(dot_west_directory)
+	execute_command(["west", "init", "-l", f"{SOF_TOP}"], cwd=west_top)
+
+def west_init_if_needed():
+	"""[summary] Validates whether west workspace had been initialized and points to SOF manifest.
+	Peforms west initialization if needed.
 	"""
 	global west_top, SOF_TOP
-	paths = [ pathlib.Path(os.getcwd()), SOF_TOP, west_top]
-	for path in paths:
-		if path.is_dir():
-			result = execute_command(["west", "topdir"], capture_output=True,
-						 text=True, check=False, cwd=path)
-		if result.returncode == 0:
-			return pathlib.Path(result.stdout.rstrip(os.linesep))
-	return None
+	west_manifest_path = pathlib.Path(SOF_TOP, "west.yml")
+	result_rootdir = execute_command(["west", "topdir"], capture_output=True, cwd=west_top,
+		timeout=10, check=False)
+	if result_rootdir.returncode != 0:
+		execute_command(["west", "init", "-l", f"{SOF_TOP}"], cwd=west_top)
+		return
+	west_root_dir = pathlib.Path(result_rootdir.stdout.decode().strip('\n')).resolve()
+	result_manifest_dir = execute_command(["west", "config", "manifest.path"], capture_output=True,
+		cwd=west_top, timeout=10, check=True)
+	west_manifest_dir = pathlib.Path(west_root_dir, result_manifest_dir.stdout.decode().strip('\n')).resolve()
+	manifest_file_result = execute_command(["west", "config", "manifest.file"], capture_output=True,
+		cwd=west_top, timeout=10, check=True)
+	returned_manifest_path = pathlib.Path(west_manifest_dir, manifest_file_result.stdout.decode().strip('\n'))
+	if not returned_manifest_path.samefile(west_manifest_path):
+		west_reinitialize(west_root_dir, returned_manifest_path)
+	else:
+		print(f"West workspace: {west_root_dir}")
+		print(f"West manifest path: {west_manifest_path}")
```

Comment:
I would print these in any case...

Reply:
Well, west init kinda prints it already and I did not want to repeat after it:
![image](https://user-images.githubusercontent.com/12482515/182353012-682c8efc-8a80-469e-9498-cbfe2e8af526.png)